internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -627,6 +623,15 @@ func setCondition(condition hivev1.ClusterDeploymentCondition, conditions *[]hiv
 	}
 }
 
+func clearClusterAPIErrors(cluster *hivev1.ClusterDeployment) {
+	conditions := &cluster.Status.Conditions
+	for i := len(*conditions) - 1; i >= 0; i-- {
```

Comment:
just wondering: why not `range` here?

Reply:
I want to iterate the conditions slice in reverse order so I could remove the indexes that indicate an api error.